### PR TITLE
Refactor EKP tests

### PR DESCRIPTION
**Pull request by ilopezgp**
# PULL REQUEST

Major refactoring of EKP unit tests.

## Purpose and Content

- Separate inverse problem definition from tests. Inverse problem definition moved to different file for readability and portability.
- Include nonlinear inverse problem test for EKI, EKS, UKI.
- Fix UKI test, which before used perfect denoised samples.
- Separate SparseEKI into a different file, and modify the test to make it much easier to pass.

### CODE REVIEW

File: test/EnsembleKalmanProcess/runtests.jl [hunk 1]
```diff
@@ -22,536 +54,362 @@ const EKP = EnsembleKalmanProcesses
...
+            y_obs,
+            Γy,
+            Inversion();
+            rng = rng,
+            failure_handler_method = IgnoreFailures(),
+            localization_method = loc_method,
         )
 
-        # UKI iterations
+        g_ens = G(get_u_final(ekiobj))
```
Comment (odunbar) [inline test/EnsembleKalmanProcess/runtests.jl:175]: transform_unconstrained_to_constrained...

File: test/EnsembleKalmanProcess/runtests.jl [hunk 2]
```diff
@@ -22,536 +54,362 @@ const EKP = EnsembleKalmanProcesses
...
-                @test_throws DimensionMismatch EKP.update_ensemble!(ukiobj, g_ens_t)
+                @test_throws DimensionMismatch EKP.update_ensemble!(ekiobj, g_ens_t)
             end
+            # Correct handling of failures
             @test !any(isnan.(params_i))
 
             # Check IgnoreFailures handler
             if i <= iters_with_failure[1]
-                params_i_unsafe = get_u_final(ukiobj_unsafe)
+                params_i_unsafe = get_u_final(ekiobj_unsafe)
```
Comment (odunbar) [inline test/EnsembleKalmanProcess/runtests.jl:213]: transform_unconstrained_to_constrained...

**Review by odunbar — APPROVED** [review]
Main comment: 
We should add 
`transform_unconstrained_to_constrained` to the any parameters that we apply "G" too, or all tests will break under any prior constraints. I've highlighted a couple of these instances

Tests all passed for me locally on 1.7.2

LGTM after addressing comments

File: test/EnsembleKalmanProcess/runtests.jl (hunk 2 above)
Comment (odunbar) [reply test/EnsembleKalmanProcess/runtests.jl:213]: etc.

File: test/SparseInversion/runtests.jl [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    prior_u2 = Dict("distribution" => Parameterized(Normal(3.0, 0.5)), "constraint" => no_constraint(), "name" => "u2")
+    prior = ParameterDistribution([prior_u1, prior_u2])
+    prior_mean = mean(prior)
+    prior_cov = cov(prior)
+
+    # Define a few inverse problems to compare algorithmic performance
+    rng_seed = 42
+    rng = Random.MersenneTwister(rng_seed)
+    lin_inv_problems =
+        [nonlinear_inv_problem(u_star, noise_level, n_obs, rng; obs_corrmat = corrmat) for corrmat in obs_corrmats]
```
[the remark this answers is not in the document]
Comment (ilopezgp) [reply test/SparseInversion/runtests.jl:42]: it was, thanks!

**Comment by ilopezgp** [comment]
Thanks for pointing out the fact that we were not testing constraint transforms. I will now add the prior transform to the inverse problem definitions, and call `G` the function from unconstrained space directly (with an implicit transform),

```
G(x) = G_linear(transform_unconstrained_to_constrained(prior, x), A)
```

**Pull request merged**